x/evm/keeper/keeper.go

```diff
@@ -285,6 +285,8 @@ func (k *Keeper) GetAccountWithoutBalance(ctx sdk.Context, addr common.Address)
 	ethAcct, ok := acct.(ethermint.EthAccountI)
 	if ok {
 		codeHash = ethAcct.GetCodeHash().Bytes()
+	} else if k.IsVirtualFrontierContract(ctx, addr) {
```

Comment:
what acct type it's for the vfc contract? it's not `(ethermint.EthAccountI)`?

Reply:
On Dymension, it is not.
It is `BaseAccount` on Dymension, thus does not contains the code hash information.

Reply:
I think the `else` is wrong. no?

Reply:
On Ethermint-based chains (eg Ethermint dev itself, RollApps,...) the default proto account used is EthAccount, so `if ok {` would match.
On Dymension, `BaseAccount` is used. `CodeHash`, which is a property of `EthAccount` does not exists, so we have to perform additional check and return the corresponding CodeHash for VFBC accounts.